logging/logging-processor/src/main/java/org/infinispan/logging/processor/report/IndentingXmlWriter.java

```diff
@@ -0,0 +1,295 @@
+/*
+ * JBoss, Home of Professional Open Source.
+ * Copyright 2016, Red Hat, Inc., and individual contributors
+ * as indicated by the @author tags. See the copyright.txt file in the
+ * distribution for a full listing of individual contributors.
+ *
+ * This is free software; you can redistribute it and/or modify it
+ * under the terms of the GNU Lesser General Public License as
+ * published by the Free Software Foundation; either version 2.1 of
+ * the License, or (at your option) any later version.
+ *
+ * This software is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public
+ * License along with this software; if not, write to the Free
+ * Software Foundation, Inc., 51 Franklin St, Fifth Floor, Boston, MA
+ * 02110-1301 USA, or see the FSF site: http://www.fsf.org.
+ */
+
+package org.infinispan.logging.processor.report;
+
+import java.util.Iterator;
+import java.util.stream.Stream;
+import javax.xml.namespace.NamespaceContext;
+import javax.xml.stream.XMLStreamConstants;
+import javax.xml.stream.XMLStreamException;
+import javax.xml.stream.XMLStreamWriter;
+
+/**
+ * IndentingXmlWriter.
+ * @author Durgesh Anaokar
+ * @since 5.3
+ */
+class IndentingXmlWriter implements XMLStreamWriter, XMLStreamConstants {
```

Comment:
Take a look at the [PrettyXMLStreamWriter](https://github.com/infinispan/infinispan/blob/1b6b6bd843cb7ac9e9e3d0f5e83265a254188f7b/commons-test/src/main/java/org/infinispan/commons/test/PrettyXMLStreamWriter.java) to see how you can simplify this. It won't be possible to reuse the class is belongs in commons, but you can reuse the code.

Reply:
This should be done before the PR is merged.